components/analytics/format.tsx

```diff
@@ -49,43 +49,22 @@ export function formatPostsPerDayData(data: any) {
             data: [] as any,
         }],
     };
-
-    for (const index in data) {
-        if (data[index]) {
-            const row = data[index];
-            chartData.labels.push(row.name);
-            chartData.datasets[0].data.push(row.value);
-        }
-    }
-
-    return chartData;
+    return fillChartData(chartData, labels, data);
 }
 
-// synchronizeChartData converges on a uniform set of labels for all entries in the given chart data. If a given label wasn't already present in the chart data, a 0-valued data point at that label is added.
-//
-// For date-labelled charts, this ensures that each charts starts and ends on the same interval, even if data for part of that interval was never collected.
-export function synchronizeChartData(...chartDatas: any[]) {
+// synchronizeChartLabels converges on a uniform set of labels for all entries in the given chart data.
+// If a given label wasn't already present in the chart data, a 0-valued data point at that label is added.
+export function synchronizeChartLabels(...datas: any) {
     const labels: Set<string> = new Set();
-
-    // collect all labels
-    chartDatas.forEach((chartData) => {
-        chartData.labels.forEach((label: string) => labels.add(label));
-    });
-
-    // fill in missing
-    chartDatas.forEach((chartData) => {
-        if (chartData.labels.length > 0) { // don't add to empty graphs
-            labels.forEach((label: string) => {
-                if (chartData.labels.indexOf(label) === -1) {
-                    chartData.labels.push(label);
-                    chartData.datasets[0].data.push(0);
-                }
-            });
+    datas.forEach((data: any) => {
+        if (data && data.length > 0) {
```

Comment:
Could this become
```suggestion
        if (data?.length) {
```
?